### ISSUE TITLE
False positive use_g_strcmp0

### ISSUE THREAD

**Issue by swick** [bilelmoussaoui/gobject-lint#48]
Saving the result to a variable should not trigger this:
```
/workspace/document-portal/permission-db.c:697:21
  ...
            /* Not in new, but in old => removed */
            remove_app_id (self, a[ia], id);
            ia++;
  ...
        else
  ...
  >          int cmp = strcmp (a[ia], b[ib]);
  ...
            if (cmp == 0)
  ...
  error: strcmp() returns 0 for equality — use 'strcmp(...) == 0' or 'strcmp(...) != 0' instead of bare boolean check
  rule: use_g_strcmp0
```

**Comment by bilelmoussaoui** [comment]
This was fixed after rules split, but I added a test case for it anyways
